src/Bicep.Core/SemanticModel/NameBindingVisitor.cs

```diff
@@ -110,30 +131,61 @@ private Symbol ValidateFunctionFlags(Symbol symbol, TextSpan span)
 
-        private Symbol LookupSymbolByName(string name, TextSpan span)
+        private Symbol LookupSymbolByName(string name, TextSpan span, string? @namespace)
         {
-            if (this.declarations.TryGetValue(name, out var localSymbol))
+            NamespaceSymbol? FindNamespace(string name) =>
+                this.namespaces
+                .Where(ns => ns.Name.Equals(name, LanguageConstants.IdentifierComparison))
+                .FirstOrDefault();
+
+            Symbol? foundSymbol;
+            if (string.IsNullOrEmpty(@namespace))
             {
-                // we found the symbol in the local namespace
-                return ValidateFunctionFlags(localSymbol, span);
+                // attempt to find name in the imported namespaces
+                var namespaceSymbol = FindNamespace(name);
+
+                if (namespaceSymbol != null)
+                {
+                    // namespace symbol found
+                    return ValidateFunctionFlags(namespaceSymbol, span);
+                }
+
+                // declarations must not have a namespace value, namespaces are used to fully qualify a function access.
+                // There might be instances where a variable declaration for example uses the same name as one of the imported
+                // functions, in this case to differentiate a variable declaration vs a function access we check the namespace value,
+                // the former case must have an empty namespace value whereas the latter will have a namespace value.
+                if (this.declarations.TryGetValue(name, out var localSymbol))
+                {
+                    // we found the symbol in the local namespace
+                    return ValidateFunctionFlags(localSymbol, span);
+                }
+
+                // attempt to find function in all imported namespaces
+                var foundSymbols = this.namespaces
+                    .Select(ns => ns.TryGetFunctionSymbol(name))
+                    .Where(symbol => symbol != null)
+                    .ToList();
+
+                if (foundSymbols.Count() > 1)
+                {
+                    // ambiguous symbol
+                    return new ErrorSymbol(DiagnosticBuilder.ForPosition(span).AmbiguousSymbolReference(name, this.namespaces.Select(ns => ns.Name)));
+                }
+
+                foundSymbol = foundSymbols.FirstOrDefault();
             }
+            else
+            {
+                // attempt to find name in the imported namespaces
+                var namespaceSymbol = FindNamespace(@namespace!);
 
-            // symbol does not exist in the local namespace
-            // try it in the imported namespaces
-
-            // match in one of the namespaces
-            var foundSymbols = this.namespaces
-                .Select(ns => ns.TryGetFunctionSymbol(name))
-                .Where(symbol => symbol != null)
-                .ToList();
+                if (namespaceSymbol == null)
```

Comment:
In the place where you're calling this function with a non-null `@namespace`, you already have the namespace symbol. Why not just pass in the namespace symbol directly, rather than looking it up again?

Reply:
The if statement handles null ```@namespace``` and does a namespace lookup based on the symbol name, this else handles non-null ```@namespace``` and does a namespace lookup based on this value. I kind of need to do two separate lookups because they use different values, is that what you mean?

Reply:
From line 102 in this file (the only place I could find where the `@namespace` arg is non-null):
```
this.LookupSymbolByName(syntax.Name.IdentifierName, syntax.Name.Span, namespaceSymbol.Name);
```

You're passing the name to this function when you already have the namespaceSymbol. Why not just update this function to take `NamespaceSymbol?` as an argument since you already have it?

Reply:
Got it, I see what you mean, makes sense.